engines/mock/doc.go

```diff
@@ -0,0 +1,20 @@
+// Package mock implements a MockEngine that doesn't really do anything, but
+// allows us to test plugins without having to run a real engine.
+package mock
+
+import (
+	"errors"
+
+	"github.com/taskcluster/taskcluster-worker/engines"
+	"github.com/taskcluster/taskcluster-worker/engines/extpoints"
+	"github.com/taskcluster/taskcluster-worker/runtime"
+)
+
+func init() {
+	// Register the mock engine as an import side-effect
+	extpoints.EngineProviders.Register(NewEngine, "mock")
```

Comment:
Actually, if we're sneaky I think we could do:
```go
func init() {
    extpoints.EngineProviders.Register(func(options *engines.EngineOptions) {
      return nil, errors.New("Engine not implemented!")
    }, "mock")
}
```

After all most engine constructors are just going to set `engines.EngineOptions` as a property on the object... and then maybe call some functions to validate that the environment it setup correclty.. ie. docker socket exists, etc...